SimPPS/RPDigiProducer/plugins/RPGaussianTailNoiseAdder.cc

```diff
@@ -0,0 +1,50 @@
+#include "SimPPS/RPDigiProducer/plugins/RPGaussianTailNoiseAdder.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include <iostream>
+#include "CLHEP/Random/RandGauss.h"
+#include "CLHEP/Random/RandFlat.h"
+#include "CLHEP/Random/RandBinomial.h"
+
+#include <cmath>
+
+using namespace std;
+
+RPGaussianTailNoiseAdder::RPGaussianTailNoiseAdder(
+    int numStrips, double theNoiseInElectrons, double theStripThresholdInE, CLHEP::HepRandomEngine &eng, int verbosity)
+    : numStrips_(numStrips),
+      theNoiseInElectrons(theNoiseInElectrons),
+      theStripThresholdInE(theStripThresholdInE),
+      rndEngine_(eng),
+      verbosity_(verbosity) {
+  strips_above_threshold_prob_ = std::erfc(theStripThresholdInE / sqrt(2.0) / theNoiseInElectrons) / 2;
+}
+
+simromanpot::strip_charge_map RPGaussianTailNoiseAdder::addNoise(const simromanpot::strip_charge_map &theSignal) {
+  simromanpot::strip_charge_map the_strip_charge_map;
+
+  // noise on strips with signal:
+  for (simromanpot::strip_charge_map::const_iterator i = theSignal.begin(); i != theSignal.end(); ++i) {
+    double noise = CLHEP::RandGauss::shoot(&(rndEngine_), 0.0, theNoiseInElectrons);
+    the_strip_charge_map[i->first] = i->second + noise;
+    if (verbosity_)
+      edm::LogInfo("RPDigiProducer") << "noise added to signal strips: " << noise << "\n";
+  }
+
+  // noise on the other strips
+  int strips_no_above_threshold =
+      CLHEP::RandBinomial::shoot(&(rndEngine_), (long)numStrips_, strips_above_threshold_prob_);
+
+  for (int j = 0; j < strips_no_above_threshold; j++) {
+    int strip = CLHEP::RandFlat::shootInt(numStrips_);
```

Comment:
@mundim @civanch why this does not use the engine?

Reply:
It scaped the attention when moving from TRandom generator to the CLHEP equivalent.